src/main/scala/com/databricks/labs/overwatch/pipeline/Schema.scala

```diff
@@ -258,11 +284,87 @@ object Schema extends SparkSessionWrapper {
+  val minimumLibrariesSchema: ArrayType = ArrayType(
+    StructType(Seq(
+      StructField("jar",StringType, nullable = true),
+      StructField("maven",
+        StructType(
+          Seq(StructField("coordinates",StringType, nullable = true)
+          )),nullable = true),
+      StructField("pypi",StructType(Seq(
+        StructField("package",StringType, nullable = true)
+      )), nullable = true),
+      StructField("whl",StringType, nullable = true)
+    )), containsNull = true
+  )
+
+  val minimumManualOverrideParamsSchema: StructType = StructType(Seq(
+    StructField("jar_params",ArrayType(StringType, containsNull = true), nullable = true),
+    StructField("python_params",ArrayType(StringType, containsNull = true), nullable = true),
+    StructField("spark_submit_params",ArrayType(StringType, containsNull = true), nullable = true),
+    StructField("notebook_params",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumSubmitRunDetailsSchema: StructType = StructType(Seq(
+    StructField("existing_cluster_id",StringType, nullable = true)
+  ))
+
+  val minimumRepairDetailsSchema: ArrayType = ArrayType(
+    StructType(Seq(
+      StructField("repair_timestamp",LongType, nullable = true),
+      StructField("rerun_tasks",StringType, nullable = true),
+      StructField("latest_repair_id",StringType, nullable = true)
+    ))
+  )
+
+  val minimumNotebookTaskSchema: StructType = StructType(Seq(
+    StructField("notebook_path",StringType, nullable = true),
+    StructField("revision_timestamp",LongType, nullable = true)
+  ))
+
+  val minimumSparkPythonTaskSchema: StructType = StructType(Seq(
+    StructField("spark_python_task",StructType(Seq(
+      StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true),
+      StructField("python_file",StringType)
+    )),nullable = true)
+  ))
+
+  val minimumSparkSubmitTaskSchema: StructType = StructType(Seq(
+    StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumSparkJarTaskSchema: StructType = StructType(Seq(
+    StructField("jar_uri",StringType,nullable = true),
+    StructField("main_class_name",StringType,nullable = true),
+    StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumShellCommandTaskSchema: StructType = StructType(Seq(
+    StructField("command",ArrayType(StringType,containsNull = true),nullable = true)
+  ))
+
+  val minimumPipelineTaskSchema: StructType = StructType(Seq(StructField("pipeline_id",StringType,nullable = true)))
+
+  val minimumPythonWheelTaskSchema: StructType = StructType(Seq(
+    StructField("entry_point",StringType, nullable = true),
+    StructField("package_name",StringType, nullable = true),
+    StructField("parameters",ArrayType(StringType, containsNull = true), nullable = true)
+  ))
+
+  val minimumSQLTaskSchema: StructType = StructType(Seq(
```

Comment:
also add dashboards & alerts

Reply:
minimum schema -- arrayType -- no risk of cardinality. When users have them, fields will appear.